### PR TITLE
improve urlParse for puppeteer-parse

### PR DESCRIPTION
**Pull request by kristol07**
**Background**: When use Android app, for links shared from some apps, links are not pure url but with additional text such as article title.
Example: "The Missing Semester of Your CS Education - https://new.ycombinator.com/item?id=34934216"

**Solution**: To improve user experience, instead of just returning parse error, it could be better to try to parse the urlStr first and check whether any valid url. So I add a check with regex before that, the code change is rather simple and understandable.

Thanks for any comments, this is my first PR.

### CODE REVIEW

**Comment by jacksonh** [comment]
Ah thank you this looks good, we have something similar in iOS 🚀 

I'll wait on @satindar to merge as he has a big branch going right now.

**Comment by satindar** [comment]
> Ah thank you this looks good, we have something similar in iOS 🚀
> 
> I'll wait on @satindar to merge as he has a big branch going right now.

I can easily rebase my branch after this is in 👍🏽

**Comment by satindar** [comment]
Thanks @kristol07 for bringing this to our attention and for this PR. We'll try to fix this client side as well. Will let @sywhb take a look here before merging.

**Comment by kristol07** [comment]
> Thanks @kristol07 for bringing this to our attention and for this PR. We'll try to fix this client side as well. Will let @sywhb take a look here before merging.

Sure, I just found one small bug by myself, sorry about that even though I actually double checked it ==

@jacksonh @satindar However, this brings another question. I didn't run it before commit, because I couldn't...
When try to build the dev environment, following steps in the current doc just doesn't work. Setting environment variables, removing some strange configuration in docker compose file... and when I finally build the frontend, I don't know how to create account and log in, so just give up for now. It could be great if we can have more detailed documentation on this.
I asked one question in discord before, but maybe buried by too many other messages. If I could run in my local machine, I'd like to contribute more as I just like it and use it everyday ;-) so here thank you again for your great effort!

File: packages/puppeteer-parse/index.js
Comment (sywhb) [inline packages/puppeteer-parse/index.js:406]: We might need to remove `g` in the regex as it won't include capture groups in matches

File: packages/puppeteer-parse/index.js
Comment (sywhb) [inline packages/puppeteer-parse/index.js:411]: We probably should use `String.prototype.match()` method since we want to fetch all the matches at one go

**Review by sywhb — CHANGES_REQUESTED** [review]

File: packages/puppeteer-parse/index.js
Comment (kristol07) [reply packages/puppeteer-parse/index.js]: Thanks for comment! yes, no need to get details of each match with loop, I updated it to use `match` method, and only return first match, what do you think?

File: packages/puppeteer-parse/index.js
Comment (kristol07) [reply packages/puppeteer-parse/index.js]: Thanks for comment, I prefer to keep `g` here, see different regex expr with different results below: 

image

I think if we need any change in the future, this expression is less likely to be updated.

Comment (kristol07) [reply packages/puppeteer-parse/index.js]: `var` is also changed to `const`

Comment (sywhb) [reply packages/puppeteer-parse/index.js]: Yeah, good point!

File: packages/puppeteer-parse/index.js
Comment (sywhb) [reply packages/puppeteer-parse/index.js]: No problem at all. I think this good to 🚢

**Review by sywhb — APPROVED** [review]
LGTM. Really appreciate @kristol07 !

**Pull request merged**
